src/Client.php

```diff
@@ -457,6 +447,28 @@ public function watch(array $pipeline = [], array $options = [])
         return $operation->execute($server);
     }
 
+    private function formatEncryptionOptions(array $options): array
+    {
+        if (isset($options['keyVaultClient'])) {
+            if ($options['keyVaultClient'] instanceof self) {
+                $options['keyVaultClient'] = $options['keyVaultClient']->manager;
+            } elseif (! $options['keyVaultClient'] instanceof Manager) {
+                throw InvalidArgumentException::invalidType('"keyVaultClient" option', $options['keyVaultClient'], [self::class, Manager::class]);
+            }
+        }
+
+        // The server requires an empty document for automatic credentials.
+        if (isset($options['kmsProviders']) && is_array($options['kmsProviders'])) {
+            foreach ($options['kmsProviders'] as $name => $provider) {
+                if ($provider === []) {
+                    $options['kmsProviders'][$name] = Document::fromPHP([]);
```

Comment:
Is it better to cast the empty array value like this?
```suggestion
                    $options['kmsProviders'][$name] = (object) $provider;
```